src/wp-includes/rest-api.php

```diff
@@ -1254,6 +1255,31 @@ function rest_add_application_passwords_to_index( $response ) {
 	return $response;
 }
 
+/**
+ * Adds the default template types and template part areas to the REST API index.
+ *
+ * @since 6.8.0
+ *
+ * @param WP_REST_Response $response REST API response.
+ * @return WP_REST_Response Modified REST API response.
+ */
+function rest_add_templates_default_data_to_index( $response ) {
+	if ( ! is_user_logged_in() ) {
+		return $response;
+	}
```

Comment:
I'd suggest a higher cap check so subscribers and low level roles added by plugins can't access the data.

Maybe allow it for any one with `edit_posts`, `edit_<registered post types>s` or theme editing permissions.